aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/RatePerSecondInstance.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2019 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.aggregation.simple;
+
+import com.spotify.heroic.ObjectHasher;
+import com.spotify.heroic.aggregation.AggregationInstance;
+import com.spotify.heroic.aggregation.AggregationOutput;
+import com.spotify.heroic.aggregation.AggregationResult;
+import com.spotify.heroic.aggregation.AggregationSession;
+import com.spotify.heroic.aggregation.BucketStrategy;
+import com.spotify.heroic.aggregation.EmptyInstance;
+import com.spotify.heroic.aggregation.RetainQuotaWatcher;
+import com.spotify.heroic.common.DateRange;
+import com.spotify.heroic.common.Series;
+import com.spotify.heroic.metric.Event;
+import com.spotify.heroic.metric.MetricCollection;
+import com.spotify.heroic.metric.MetricGroup;
+import com.spotify.heroic.metric.MetricType;
+import com.spotify.heroic.metric.Payload;
+import com.spotify.heroic.metric.Point;
+import com.spotify.heroic.metric.Spread;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Data;
+
+/**
+ * A rate per second implementation inspired by the prometheus irate() function.
+ *
+ * It is intended to be be used with counters that are increasing to calculate the rate of change
+ * per second difference between two points. It handles the counter being reset.
+ *
+ * @author dmichel
+ */
+@Data
+public class RatePerSecondInstance implements AggregationInstance {
```

Comment:
Do we need a special case for this? I think having both will be confusing for people trying to decide which one to use. What are the different use cases you're picturing for someone to want rate vs rate per second?

Reply:
I think we do - this is similar to the delta and deltaPerSecond aggregations we have already.

for SLO calculates over long time periods I envision using just the rate aggregation to get a total sum.